[7.0] new module: record_archiver

guewen opened the pull request:
New module _record_archiver_.
This module allows to configure a lifespan for different models. When this lifespan is reached, the records are archived (with some exceptions such as records not being `done` or `canceled`).
The default archiving strategy is to set the `active` flag to `False`.

yvaucher committed:
Add module rusty_record_reaper_runner

Commit:
Rename module

From rusty_record_reaper_runner to record_archiver.
Not that I dislike the old funny name, but I guess it will be easier to get
what it does with the new (maybe too serious) name.

Commit:
Make the archiver more extensible

by extracting methods. Now we are able to extend the module to implement other
archiving stategies.

Commit:
Use a m2o for the models

Commit:
Hide models without an active field

Commit:
Remove the company on record.lifespan

The company was mandatory and prevented to use the module for models without
company_id.  Also, there is currently no use case for a different configuration
for different companies.  It simplifies the setup too.

If a use case exists for a different setup per company, it can be reintroduced
later with a optional company_id.

Commit:
Add tests an fix a condition

On the search of active models

Commit:
Add access rights

Commit:
Add tests for the archiving of records

Commit:
Add translation pot

gurneyalex commented:
could you rebase your PR to benefit from the recent runbot fixes?

guewen commented:
@gurneyalex done

yvaucher commented:
It has lost it's alliteration :sob: 

:+1: Thanks for the tests and cleaning

It needs a separation in travis tests with the module adding `notification_email_send`

```
IntegrityError: null value in column "notification_email_send" violates not-null constraint
```

lmignon commented on record_archiver/__openerp__.py:
```diff
@@ -0,0 +1,47 @@
+
+Lifespan is defined per record per company.
+ """,
+ 'author': 'Camptocamp',
```
OCA is not listed in the list of authors

lmignon commented on record_archiver/__openerp__.py:
```diff
@@ -0,0 +1,47 @@
+{'name': 'Records Archiver',
+ 'version': '0.1',
+ 'description': """
+Create a cron job that deactivates old records in order to optimize
```
It would be nice to format the description with the same rules as for the 8.0 branch

Commit:
Use the OCA template in __openerp__.py

guewen commented:
Rebased.

@lmignon you are right, fixed

> It needs a separation in travis tests with the module adding notification_email_send
> 
> IntegrityError: null value in column "notification_email_send" violates not-null constraint

:anguished: it is in the `mail` addon.

yvaucher commented:
It is the same issue as described here:

https://github.com/OCA/server-tools/blob/8.0/auth_from_http_remote_user/tests/test_res_users.py#L72-L80

yvaucher commented:
@lmignon as you are reviewing this PR you can comment on this error in travis? You know what cause this?

Commit:
Exclude record_archiver from the one-by-one tests

Because an error prevent creation of res.partner records in the tests due the
'mail' addon adding a required 'notification_email_send' field. This addon does
not depend on 'mail' so the test violates the not-null constraint.

The tests are passing in the normal builds anyway.

guewen commented:
My last commit excludes this module from the one-by-one tests. This error is very weird though because it shows that modules are impacting each other in these `UNIT_TESTS` tests which should exactly not be the case.

yvaucher commented:
:+1:

lmignon commented:
@yvaucher I've tried a lot of way to turn around this bug but failed to find the right solution. That's why I've added the `@unittest.skipIf(os.environ.get('TRAVIS'))` decorator in https://github.com/OCA/server-tools/blob/8.0/auth_from_http_remote_user/tests/test_res_users.py#L72-L80

lmignon commented:
:+1:

yvaucher commented:
@lmignon Thanks ok here @guewen disable it only when all unit tests are launched.

gurneyalex commented:
merging
